upcoming: [DI-23436] - Integrating the Resources section in the create alert flow for Cloudpulse alerting

venkymano-akamai opened the pull request:
## Description 📝

Integrating the Resources section in the create alert flow for Cloudpulse alerting

## Changes  🔄

1. Add new component CloudPulseModifyAlertResources that consumes the common resources section in a form.
2. Enhance the create alert page form to use the CloudPulseModifyAlertResources for capturing the selected entity ids

## Target release date 🗓️
25-02-2025

## Preview 📷

**Include a screenshot or screen recording of the change.**

:lock: Use the [Mask Sensitive Data](https://cloud.linode.com/profile/settings) setting for security.

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
|![Screenshot 2025-02-12 at 2 58 25 PM](https://github.com/user-attachments/assets/a7bae0eb-62ad-4a41-aabd-bcb286a93cf6)|![Screenshot 2025-02-12 at 2 55 57 PM](https://github.com/user-attachments/assets/150d774f-730e-49fa-846b-1faf6b427e60)|
|![Screenshot 2025-02-12 at 2 56 05 PM](https://github.com/user-attachments/assets/30c7ac95-cf70-4d1b-a870-649e88e09a1d)|![Screenshot 2025-02-12 at 2 56 11 PM](https://github.com/user-attachments/assets/b468d8ea-9237-4955-b0e5-8036485d9dd7)|
|![Screenshot 2025-02-12 at 2 56 18 PM](https://github.com/user-attachments/assets/92ef76ba-8f0a-46a5-b85a-ab7383fb4daa)|![Screenshot 2025-02-12 at 2 55 24 PM](https://github.com/user-attachments/assets/41104c61-89b4-4c50-9359-6a286ce52f7d)|

## How to test 🧪

1. Login as mock user as some endpoints are not available.
2. Navigate to the monitor tab and then to alerts tab
3. Click on the Create Alert button
4. You will be able to see the create alert page with the resources section and able to select and publish the entity_ids

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>

- [ ] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [ ] All unit tests are passing
- [ ] TypeScript compilation succeeded without errors
- [ ] Code passes all linting rules

</details>

venkymano-akamai commented on packages/manager/src/features/CloudPulse/Alerts/CreateAlert/schemas.ts:
```diff
@@ -3,16 +3,8 @@ import { object, string } from 'yup';
-
 export const CreateAlertDefinitionFormSchema = createAlertDefinitionSchema.concat(
   object({
-    engineType: engineOptionValidation,
-    region: string().required(fieldErrorMessage),
```
FYI, these validations are not needed now

santoshp210-akamai commented on packages/manager/src/features/CloudPulse/Alerts/AlertsResources/AlertsResources.tsx:
```diff
@@ -279,10 +299,12 @@ export const AlertResources = React.memo((props: AlertResourcesProp) => {
-        {alertLabel || 'Resources'}
-        {/* It can be either the passed alert label or just Resources */}
-      </Typography>
+      {!hideLabel && (
+        <Typography ref={titleRef} variant="h2">
+          {alertLabel || 'Resources'}
+          {/* It can be either the passed alert label or just Resources */}
+        </Typography>
+      )}
```
Everything looks good!. 
Nit: We are using the same lines of code at Line 283-288.

venkymano-akamai commented on packages/manager/src/features/CloudPulse/Alerts/AlertsResources/AlertsResourcesNotice.tsx:
```diff
@@ -38,22 +40,29 @@ export const AlertsResourcesNotice = React.memo(
         </Typography>
         {isSelectAll && (
-          <StyledLinkButton
+          <LinkButton
```
FYI, here we are just fine with link button only

jaalah-akamai commented:
We'll get a code review on this work soon!
cc: @carrillo-erik for code review

mjac0bs commented on packages/manager/src/features/CloudPulse/Alerts/AlertsResources/AlertsResources.tsx:
We may want to have a look at the layout of the Show Selected checkbox in this section, as it renders overlapping the region select when a selected region has a very long name. (Though, this may not be a real issue in prod because I don't think any region names are that long, currently.)

![Screenshot 2025-02-12 at 2 00 47 PM](https://github.com/user-attachments/assets/9b5d66a7-b834-4b7a-b7d5-929c50c77b01)

carrillo-erik replied:
This might need further adjustment with the vertical alignment.

![Screenshot 2025-02-13 at 8 17 15 AM](https://github.com/user-attachments/assets/e527d2da-e3af-4749-b492-27b684a3b9c5)

venkymano-akamai replied:
@carrillo-erik , this is also a common component underneath, it is happening only for gecko region, and if i select other regions it is working, this needs fix in common component is what i feel.

kmuddapo replied:
@jaalah-akamai   Any thoughts here?

mjac0bs commented on packages/manager/src/features/CloudPulse/Alerts/AlertsResources/AlertsResources.tsx:
```diff
@@ -343,6 +365,18 @@ export const AlertResources = React.memo((props: AlertResourcesProp) => {
             </Grid>
           )}
         </Grid>
+        {noSelectionErrorText && (
+          <Grid item xs={12}>
+            <Typography
+              sx={(theme) => ({
+                color: theme.tokens.content.Text.Negative,
+              })}
+              variant="body2"
+            >
+              {noSelectionErrorText}
+            </Typography>
+          </Grid>
+        )}
```
I'm curious if UX considered an error notice for this section, rather than the Typography error text that's not associated with a specific field.

Here's a couple of example of what we do elsewhere:
![Screenshot 2025-02-12 at 2 10 47 PM](https://github.com/user-attachments/assets/8b555462-d2eb-4ab7-ab6f-decb13f3d98d)



![Screenshot 2025-02-12 at 2 08 52 PM](https://github.com/user-attachments/assets/872d68b3-780e-42fa-a966-e232faa48f4b)

venkymano-akamai replied:
> I'm curious if UX considered an error notice for this section, rather than the Typography error text that's not associated with a specific field.
> 
> Here's a couple of example of what we do elsewhere: ![Screenshot 2025-02-12 at 2 10 47 PM](https://private-user-images.githubusercontent.com/114685994/412632686-8b555462-d2eb-4ab7-ab6f-decb13f3d98d.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkzOTg3NjksIm5iZiI6MTczOTM5ODQ2OSwicGF0aCI6Ii8xMTQ2ODU5OTQvNDEyNjMyNjg2LThiNTU1NDYyLWQyZWItNGFiNy1hYjZmLWRlY2IxM2YzZDk4ZC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxMlQyMjE0MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZjRiOWE3MDU1YmJiODVhYTUwZmYxM2NhNTY3ZmFlZTM4Mjc3NTM4OWRkYmEzZmQ2YzhlNzM0MmI0ZjFmOTYyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.AecJpSlajME75GZlQb-RQPsyc3gECx_gThxX_Lawo1o)
> 
> ![Screenshot 2025-02-12 at 2 08 52 PM](https://private-user-images.githubusercontent.com/114685994/412632680-872d68b3-780e-42fa-a966-e232faa48f4b.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkzOTg3NjksIm5iZiI6MTczOTM5ODQ2OSwicGF0aCI6Ii8xMTQ2ODU5OTQvNDEyNjMyNjgwLTg3MmQ2OGIzLTc4MGUtNDJmYS1hOTY2LWUyMzJmYWE0OGY0Yi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxMlQyMjE0MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZWM3NDNiYjAxOGUyZDA3NzM1YzcxOWFmOGI4ZWU3MzFmMjM1ZGEyODUzZWQyMmE3NTRhZjUzMmY5OWQ4ZmE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ZnnFWBJ4LYObnyM8ix7xm7-KAsVb0sXt6srTp7_V4MA)

@mjac0bs , this is the figma given by UX for error state

https://www.figma.com/design/26kXnWVNU5tK2xQdDdISuC/Custom-Alerts?node-id=3874-75449&t=WiHlnXDAndjFdovu-0#1103605963

kmuddapo replied:
This was proposed by UX, let me discuss this with UX and get back if we can consider this feedback. Thanks!!

kmuddapo replied:
As per the latest internal conversation,   We will allow alert to get created with zero/empty resources as well , In that case this error validation is not required. Lets remove this validation.  Thanks!!

venkymano-akamai replied:
@kmuddapo , Thanks, updated the same, removed validations, now empty array will go in entity_ids if no resources is selected

mjac0bs reviewed (commented):
Thanks @venkymano-akamai. 

I'm not seeing this in the ACLP - Contextual Views Figma mock, where I see Resources represented in the an alert details view, but not a create alert view, so I left a few questions about the UI presentation.

Unrelated to the changes in this PR, but noticed while testing, I believe we have a typo in the Services dropdown, where `Linode` should be plural (`Linodes`) to match `Databases`.

![Screenshot 2025-02-12 at 1 49 38 PM](https://github.com/user-attachments/assets/c787f516-a140-4abd-96c7-ca67b577bbf6)

venkymano-akamai commented on packages/manager/src/features/CloudPulse/Alerts/AlertsResources/AlertsResourcesNotice.tsx:
```diff
@@ -38,22 +40,26 @@ export const AlertsResourcesNotice = React.memo(
-          <StyledLinkButton
+          <LinkButton
+            onClick={() => {
+              handleSelectionChange('Unselect All');
+            }}
             aria-label="Unselect All Resources"
             data-testid="unselect_all_notice"
-            onClick={() => handleSelectionChange('Unselect All')}
           >
             Unselect All
-          </StyledLinkButton>
+          </LinkButton>
```
> 1. What's the reasoning behind including this select/unselect all button in the info notice? The user has the ability to select or unselect via the checkbox in the table down below.
> 
> ![Screenshot 2025-02-12 at 1 40 52 PM](https://private-user-images.githubusercontent.com/114685994/412624243-622bd8eb-8407-4264-afad-a3303d8feae7.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk0MDc3MzgsIm5iZiI6MTczOTQwNzQzOCwicGF0aCI6Ii8xMTQ2ODU5OTQvNDEyNjI0MjQzLTYyMmJkOGViLTg0MDctNDI2NC1hZmFkLWEzMzAzZDhmZWFlNy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxM1QwMDQzNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01Mzk5ZjQwMTMxOWY5YjJlMmJlMzE3YTVjOTMwZTYzNGI3YjM1ZTkzZTk1MTIwYmI5ZjQwZTg3Njg4YjNiYTc1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.CSSZfKN3a2Acrqjg310QSEWl2bxMIIR3WM2A7MqiMyQ)
> 
> 2. Regarding the select/unselect all checkbox in the table, I'm noticing the 'unselect' box is a smaller size. Can that checkbox size be set consistently in the `DisplayAlertResources.tsx`table?
> 
>  Screen.Recording.2025-02-12.at.1.25.42.PM.mov

@mjac0bs , The select all and unselect all in notice is different from table

1. Select all and unselect all in notice selects the overall resources across the pages
2. Where as the root checkbox in table selects only the resources in the first page / respective page of the table

Regarding the size of the checkbox in the indeterminate state, it is the default behaviour of the checkbox, as we are not controlling any height or width there, do you really think we need to modify that?

carrillo-erik replied:
I tested the behavior for the checkbox in question and what I noticed is that when all the checkboxes are either Selected or Unselected, then the size remains consistent with the other checkboxes. The size discrepancy is only visible when there is a combination of Selected and Unselected checkboxes. It's worth inspecting how other pages in the application treat this behavior for better understanding.

**Select All:**
![Screenshot 2025-02-13 at 8 02 25 AM](https://github.com/user-attachments/assets/26864199-838b-48ed-a9b4-8e43d31061d4)

**Unselect All:**
![Screenshot 2025-02-13 at 8 09 19 AM](https://github.com/user-attachments/assets/9ab29f30-7e29-4836-b364-281a5e7799f0)

**Combination of Selected and Unselected:**
![Screenshot 2025-02-13 at 8 02 08 AM](https://github.com/user-attachments/assets/056aae30-62cd-47ef-970a-031e21099850)

kmuddapo replied:
Thats the default behavior for the indeterminate state( combination of select +unselect).

As discussed in todays sync up ( @coliu-akamai )  Checkbox size shrinking, [DI-23478] has been raised to analyze and see if we can update the default behavior and make it consistent in the next iteration.  Hope this shouldn't block this PR for further approvals and merge. Thanks!!

venkymano-akamai replied:
@carrillo-erik , i don't see any handling for indeterminate state in pages, we have created a ticket to track this and fix for now

mjac0bs replied:
> @mjac0bs , The select all and unselect all in notice is different from table
> 
> 1. Select all and unselect all in notice selects the overall resources across the pages
> 2. Where as the root checkbox in table selects only the resources in the first page / respective page of the table

Got it - I see what you mean and how both have their uses. Thanks for clarifying.

> Regarding the size of the checkbox in the indeterminate state, it is the default behaviour of the checkbox, as we are not controlling any height or width there, do you really think we need to modify that?

Yes, it's a noticeable user-facing bug that we wouldn't want to knowingly ship to prod. It sounds like the issues has already been raised (thanks @coliu-akamai) and DI-23478 created - thanks for looking into this in the future.

venkymano-akamai replied:
@mjac0bs / @carrillo-erik  / @cliu-akamai , Also, there is a custom icon we can pass **indeterminateIcon** for this state using a prop of checkbox, since this is linode/ui component, we can create a custom icon and fix it there for indeterminate state is what i feel.

coliu-akamai replied:
nonblocking: Regarding Mariah's first point about the select/unselect all button in the info notice, I do see how it could be a bit confusing despite their different uses. Maybe even adding the word 'total' to the notice could help? Like:

![image](https://github.com/user-attachments/assets/c7c35275-f70d-49cd-ae11-cc11c9eac682)

venkymano-akamai commented:
> Thanks @venkymano-akamai.
> 
> I'm not seeing this in the ACLP - Contextual Views Figma mock, where I see Resources represented in the an alert details view, but not a create alert view, so I left a few questions about the UI presentation.
> 
> Unrelated to the changes in this PR, but noticed while testing, I believe we have a typo in the Services dropdown, where `Linode` should be plural (`Linodes`) to match `Databases`.
> 
> ![Screenshot 2025-02-12 at 1 49 38 PM](https://private-user-images.githubusercontent.com/114685994/412626907-c787f516-a140-4abd-96c7-ca67b577bbf6.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk0MDc3MzgsIm5iZiI6MTczOTQwNzQzOCwicGF0aCI6Ii8xMTQ2ODU5OTQvNDEyNjI2OTA3LWM3ODdmNTE2LWExNDAtNGFiZC05NmM3LWNhNjdiNTc3YmJmNi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxM1QwMDQzNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYWFjNzA0YzM2YzViNjNjY2ZhMjg1M2RjMzYyYjE3M2VkYjMxYzMzMWU4NWFhNjE5N2QwZWYzZTg2YWE0YWU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._N77d12eECEv8N4hvndgXafYDq3OPetDk9KceZOdIF4)

@mjac0bs , https://www.figma.com/design/26kXnWVNU5tK2xQdDdISuC/Custom-Alerts?node-id=3874-80353&p=f&t=Re7FIFpVAE712Rbo-0

This is the figma where it has the resources section in the create alert flow

venkymano-akamai commented:
@mjac0bs , Thanks for the review, Addressed couple of comments, and replied back to some of the comments

Merging my replies into single comment

1. Figma where resources section is integrated in create flow - https://www.figma.com/design/26kXnWVNU5tK2xQdDdISuC/Custom-Alerts?node-id=3874-80353&p=f&t=ICgoIKfNR3BQBenh-0
2. Figma where error state is shown - https://www.figma.com/design/26kXnWVNU5tK2xQdDdISuC/Custom-Alerts?node-id=3958-46614&t=WiHlnXDAndjFdovu-0
3. The checkbox indeterminate state size is a default behaviour and lets have it.
4. Fixed to show the notice only if resources size is greater than 0
5. Changed Linode to Linodes to maintain consistency
6. Yes i have removed the regions validation since we need to select at least one entity.
7. The select all and unselect all is different from notice

**1. Select all and unselect all in notice selects the overall resources across the pages
2. Where as the root checkbox in table selects only the resources in the first page / respective page of the table**

Please see if this is okay for approval / any other places to be fixed

kmuddapo commented:
> @mjac0bs , Thanks for the review, Addressed couple of comments, and replied back to some of the comments
> 
> Merging my replies into single comment
> 
> 1. Figma where resources section is integrated in create flow - https://www.figma.com/design/26kXnWVNU5tK2xQdDdISuC/Custom-Alerts?node-id=3874-80353&p=f&t=ICgoIKfNR3BQBenh-0
> 2. Figma where error state is shown - https://www.figma.com/design/26kXnWVNU5tK2xQdDdISuC/Custom-Alerts?node-id=3958-46614&t=WiHlnXDAndjFdovu-0 , @mjac0bs , this is not applicable, we don't need one resource required, we can create alert with empty resources, now if we don't select a resource, entity_ids will go as empty array
> 3. The checkbox indeterminate state size is a default behaviour and lets have it.
> 4. Fixed to show the notice only if resources size is greater than 0
> 5. Changed Linode to Linodes to maintain consistency
> 6. Yes i have removed the regions validation since we need to select at least one entity.
> 7. The select all and unselect all is different from notice
> 8. Moved the show selected only out of the filters section and placed it below to avoid overlapping
> 
> **1. Select all and unselect all in notice selects the overall resources across the pages 2. Where as the checkbox in the header in table selects only the resources in the first page / respective page of the table**
> 
> Please see if this is okay for approval / any other places to be fixed


@coliu-akamai   As discussed For #3 - Checkbox size shrinking, [DI-23478] has been raised to analyze and see if we can update the default behavior and make it consistent in the next iteration. Thanks!!

mjac0bs commented on packages/validation/src/cloudpulse.schema.ts:
Can you please run `yarn changeset` and add a "Removed" changeset to the validation package? (e.g. "Removed required `entity_ids` from `createAlertDefinitionSchema`"?)

venkymano-akamai replied:
Done

mjac0bs reviewed (approved):
@venkymano-akamai Thanks for addressing feedback. Approving pending a changeset is added for the update to the validation package.

coliu-akamai reviewed (commented):
Thanks for making the follow up ticket! 

I found a few other things (listed below) in the existing panels that are unrelated to this PR - since they're unrelated, could you make follow up tickets for them as well? (alternatively, if they're quick enough to handle in this PR, that would be awesome). 

This PR has the necessary amount of approvals, so after making the follow up tickets or addressing them here, feel free to merge (and after addressing Mariah's comment/any other outstanding comments above ^).

1) The Add Dimension Filter button has a very large width, so clicking to the far right still adds another dimension:
![image](https://github.com/user-attachments/assets/fe6ab29c-4e64-40f8-aed9-ebdc7b044392)

2) Is percent misaligned?
![image](https://github.com/user-attachments/assets/3a167360-30bd-4f66-ab6a-2c60b483a6b8)
![image](https://github.com/user-attachments/assets/7afa43e4-58ce-476c-9ca7-41fff07bf262)

3) For trigger conditions, the text shifts when there's an error:
![image](https://github.com/user-attachments/assets/d113b8d8-9fd7-4dc4-a283-6f4c8d9e457b)
![image](https://github.com/user-attachments/assets/0377e00c-22ca-42da-9e3b-1ade2f12fdc4)

venkymano-akamai commented:
> Thanks for making the follow up ticket!
> 
> I found a few other things (listed below) in the existing panels that are unrelated to this PR - since they're unrelated, could you make follow up tickets for them as well? (alternatively, if they're quick enough to handle in this PR, that would be awesome).
> 
> This PR has the necessary amount of approvals, so after making the follow up tickets or addressing them here, feel free to merge (and after addressing Mariah's comment/any other outstanding comments above ^).
> 
> 1. The Add Dimension Filter button has a very large width, so clicking to the far right still adds another dimension:
>    ![image](https://private-user-images.githubusercontent.com/139280159/413051251-fe6ab29c-4e64-40f8-aed9-ebdc7b044392.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk0OTAyMDUsIm5iZiI6MTczOTQ4OTkwNSwicGF0aCI6Ii8xMzkyODAxNTkvNDEzMDUxMjUxLWZlNmFiMjljLTRlNjQtNDBmOC1hZWQ5LWViZGM3YjA0NDM5Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxM1QyMzM4MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYjg2M2YwNTYzMGEyNmExMjdlNjQ4Y2E2Mjg4ZDllYmFlYTUxNGIxYzEwOGZkOGE0NTJiN2UwZWNkMGEwYjA0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.YyaB1dqyWiI1s4yzgmR4afy8tF6-6knvUyUGjjONC7Y)
> 2. Is percent misaligned?
>    ![image](https://private-user-images.githubusercontent.com/139280159/413051873-3a167360-30bd-4f66-ab6a-2c60b483a6b8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk0OTAyMDUsIm5iZiI6MTczOTQ4OTkwNSwicGF0aCI6Ii8xMzkyODAxNTkvNDEzMDUxODczLTNhMTY3MzYwLTMwYmQtNGY2Ni1hYjZhLTJjNjBiNDgzYTZiOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxM1QyMzM4MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYmZjMGI5OWZiYWY2OWJiNGYwMDljODAxZGYwYTkwZGU3ZWQ5OGFhZDBmODJiZmFlY2Q5N2FmYmU0Yzc2YTgwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.7gM-AJdh6L-VsyKGPxTRds1nbuW22UO9n-q_Zaz_Edw)
>    ![image](https://private-user-images.githubusercontent.com/139280159/413052371-7afa43e4-58ce-476c-9ca7-41fff07bf262.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk0OTAyMDUsIm5iZiI6MTczOTQ4OTkwNSwicGF0aCI6Ii8xMzkyODAxNTkvNDEzMDUyMzcxLTdhZmE0M2U0LTU4Y2UtNDc2Yy05Y2E3LTQxZmZmMDdiZjI2Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxM1QyMzM4MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MzdiYmU4ZDE0YmMwYWRiY2JhZGJmN2Y1ZGM0N2M5YzIxN2JhZTliZjQ0ZjkzZGNlNjNlNzk0MGZmOTUzZDJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.QYopRSM4YeBWfzUfKVfL7uWJWTXRYC90QUYzvqgeizs)
> 3. For trigger conditions, the text shifts when there's an error:
>    ![image](https://private-user-images.githubusercontent.com/139280159/413052128-d113b8d8-9fd7-4dc4-a283-6f4c8d9e457b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk0OTAyMDUsIm5iZiI6MTczOTQ4OTkwNSwicGF0aCI6Ii8xMzkyODAxNTkvNDEzMDUyMTI4LWQxMTNiOGQ4LTlmZDctNGRjNC1hMjgzLTZmNGM4ZDllNDU3Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxM1QyMzM4MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04OWQ3MjYwNTUzYTFkMDcyNTY4N2I3NzFlYzIxMDUxNzg5Y2U4ZjlhMWMxMzQzMDM4ZDFlYzdjODZmY2M2ODc0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.KBGy--wY0szxfdlVo52ir3Nr_CSpPucLnooxNGjJ388)
>    ![image](https://private-user-images.githubusercontent.com/139280159/413052208-0377e00c-22ca-42da-9e3b-1ade2f12fdc4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk0OTAyMDUsIm5iZiI6MTczOTQ4OTkwNSwicGF0aCI6Ii8xMzkyODAxNTkvNDEzMDUyMjA4LTAzNzdlMDBjLTIyY2EtNDJkYS05ZTNiLTFhZGUyZjEyZmRjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxM1QyMzM4MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNTQ3MzEyOTU0NGNhNWFkMGRiOGY5MzBjYzJkNmQyZGY5NTk2OTJlMjhjMTIxZTY2MzQyN2M2OWY0ZGQyMGY3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Vj6l5yCwcIUo4i6Htacd1aLiRxzITxroBxrK3hwY_do)

@coliu-akamai , we already have a UX ticket to work on next sprint added these points there in the description https://track.akamai.com/jira/browse/DI-23319

venkymano-akamai commented:
Merging this since, tests pass, jobs pass, linting and UT's pass ,all cypress pass after a very very small fix.
